kubernetes: Mark cilium-etcd-secret secret required

tgraf opened the pull request:
With the etcd-operator becoming the standard, marking the cilium-etcd-secrets as optional is inefficient as it requires a Cilium pod restart to mount the secret after it has been created.

Instaead, mark the secret as required an provide a new `cilium-external-etcd.yaml` which contains an empty secret. For users upgrading, document the requirement to create an empty secret before the upgrade.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6718)
<!-- Reviewable:end -->

aanm commented on Documentation/install/upgrade.rst:
```diff
@@ -268,6 +268,17 @@ Annotations:
+#. **Action required:** Cilium 1.4.0 marks the secret ``cilium-etcd-secrets``
+   as required in the new DaemonSet. This is done to simplify the default
+   installation with the etcd-operator. If you are already using the
+   ``cilium-etcd-secretes`` to access etcd using TLS, no action is required. If
```
`s/cilium-etcd-secretes/cilium-etcd-secrets/`

aanm commented on Documentation/install/upgrade.rst:
```diff
@@ -268,6 +268,17 @@ Annotations:
+   installation with the etcd-operator. If you are already using the
+   ``cilium-etcd-secretes`` to access etcd using TLS, no action is required. If
+   you have not been using TLS yet, you need to create an empty
+   ``cilium-etcd-secrets`` like this:
```
Write an example how can the user know if it has secrets installed or not

aanm reviewed (changes requested):
Why don't we make the creation of the dummy secret part of the upgrade guide? Regardless if the user is running secrets or not they will either *not* be overwritten or created.

```
kubectl create -f \ |SCM_WEB|\/examples/kubernetes/1.11/etcd-secrets.yaml
Error from server (AlreadyExists): error when creating "examples/kubernetes/1.13/secret.yaml": secrets "cilium-etcd-secrets" already exists
kubectl apply -f ./cilium-ds.yaml
...
```

aanm reviewed (commented):
@tgraf New thoughts around this

1) by marking the secret as mandatory this will make `cilium-etcd-operator` impossible to work without `hostNetwork: true`.
2) It seems `cilium-etcd-operator` didn't start in [1] due the fact the node is marked as nonschedulable or even that kubelet can not create the `cilium-etcd-operator` as `cilium` is not initialized and `cilium` will not be initialized because `cilium-etcd-operator` did not create the secret.

[1] https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/9272/execution/node/83/log/

tgraf committed:
kubernetes: Mark cilium-etcd-secret secret required

With the etcd-operator becoming the standard, marking the cilium-etcd-secrets
as optional is inefficient as it requires a Cilium pod restart to mount the
secret after it has been created.

Instaead, mark the secret as required an provide a new
`cilium-external-etcd.yaml` which contains an empty secret. For users
upgrading, document the requirement to create an empty secret before the
upgrade.

Signed-off-by: Thomas Graf <thomas@cilium.io>
